internal/output/html.go

```diff
@@ -0,0 +1,486 @@
+		}
+
+		// Get the max severity from groupInfo and increase the count
+		vuln.Summary.SeverityScore = groupInfo.MaxSeverity
+		vuln.Summary.SeverityRating, _ = severity.CalculateRating(vuln.Summary.SeverityScore)
+
+		if _, isUncalled := uncalledVulnIDs[vuln.Summary.ID]; isUncalled {
+			packageResult.UncalledVulns = append(packageResult.UncalledVulns, vuln)
+			packageResult.HTMLVulnCount.Uncalled = len(packageResult.UncalledVulns)
+
+			continue
+		}
+
+		packageResult.CalledVulns = append(packageResult.CalledVulns, vuln)
+		packageResult.HTMLVulnCount.Called = len(packageResult.CalledVulns)
+		addCount(&packageResult.HTMLVulnCount, vuln.Summary.SeverityRating)
+		if vuln.Summary.FixedVersion == UNFIXED {
+			packageResult.HTMLVulnCount.UnFixed += 1
+		} else {
+			packageResult.HTMLVulnCount.Fixed += 1
+		}
+	}
+
+	results := make([]PackageResult, 0, len(packageResults))
+	for _, result := range packageResults {
+		if len(result.CalledVulns) > 0 {
+			result.InstalledVersion = result.CalledVulns[0].Summary.InstalledVersion
+			result.FixedVersion = getMaxFixedVersion(ecosystem, result.CalledVulns)
+		}
+
+		results = append(results, *result)
+	}
+
+	return results
+}
+
+// processVulnerabilities processes vulnerabilities for a package
+// and returns a slice of HTMLVulnResult.
+func processVulnerabilities(vulnPkg models.PackageVulns) []HTMLVulnResult {
+	vulnResults := make([]HTMLVulnResult, len(vulnPkg.Vulnerabilities))
+	for i, vuln := range vulnPkg.Vulnerabilities {
+		aliases := strings.Join(vuln.Aliases, ", ")
+		vulnDetails := map[string]string{
+			"aliases":     aliases,
+			"description": vuln.Details,
+		}
+		if vulnPkg.Package.ImageOrigin != nil {
+			vulnDetails["layerCommand"] = vulnPkg.Package.ImageOrigin.OriginCommand
+			vulnDetails["layerId"] = vulnPkg.Package.ImageOrigin.LayerID
+			vulnDetails["inBaseImage"] = strconv.FormatBool(vulnPkg.Package.ImageOrigin.InBaseImage)
+		}
+
+		fixedVersion := getFixVersion(vuln.Affected, vulnPkg.Package.Version, vulnPkg.Package.Name, models.Ecosystem(vulnPkg.Package.Ecosystem))
+
+		vulnResults[i] = HTMLVulnResult{
+			Summary: HTMLVulnResultSummary{
+				ID:               vuln.ID,
+				PackageName:      vulnPkg.Package.Name,
+				InstalledVersion: vulnPkg.Package.Version,
+				FixedVersion:     fixedVersion,
+			},
+			Detail: vulnDetails,
+		}
+	}
+
+	return vulnResults
+}
+
+// processVulnerabilityGroups processes vulnerability groups and IDs,
+// populating the called and uncalled maps.
+func processVulnerabilityGroups(groupIDs map[string]models.GroupInfo, vulnPkg models.PackageVulns, uncalledVulnIDs map[string]bool, calledPackages map[string]bool, uncalledPackages map[string]bool) {
+	for _, group := range vulnPkg.Groups {
+		slices.SortFunc(group.IDs, identifiers.IDSortFunc)
+		representID := group.IDs[0]
+		groupIDs[representID] = group
+
+		if !group.IsCalled() {
+			uncalledVulnIDs[representID] = true
+			uncalledPackages[vulnPkg.Package.Name] = true
```

Comment:
I'm not sure if this is correct? 

If the package has one vuln that is called, one that is not called, this will count the package as an uncalled package.

Reply:
I think this would count that package as both uncalled and called. We record the called vulnerability information by each vulnerability, and this loop through each vuln and update the package count.

Reply:
Hmm... I think if a package has any called vulns, it should be categorised as called package.

Reply:
Discussed offline